core-plugins/src/test/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSourceTest.java

```diff
@@ -0,0 +1,421 @@
+import co.cask.cdap.proto.artifact.AppRequest;
+import co.cask.cdap.proto.artifact.ArtifactSummary;
+import co.cask.cdap.proto.id.ArtifactId;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.test.ApplicationManager;
+import co.cask.cdap.test.DataSetManager;
+import co.cask.cdap.test.MapReduceManager;
+import co.cask.cdap.test.TestConfiguration;
+import co.cask.hydrator.common.Constants;
+import com.google.common.base.Charsets;
+import org.apache.commons.io.FileUtils;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.python.google.common.collect.ImmutableMap;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Unit test for {@link XMLReaderBatchSource} class.
+ */
+public class XMLReaderBatchSourceTest extends HydratorTestBase {
+
+  private static final ArtifactVersion CURRENT_VERSION = new ArtifactVersion("3.4.0-SNAPSHOT");
+  private static final ArtifactId BATCH_APP_ARTIFACT_ID =
+    NamespaceId.DEFAULT.artifact("etlbatch", CURRENT_VERSION.getVersion());
+  private static final ArtifactSummary ETLBATCH_ARTIFACT =
+    new ArtifactSummary(BATCH_APP_ARTIFACT_ID.getArtifact(), BATCH_APP_ARTIFACT_ID.getVersion());
+  private static final String SOURCE_FOLDER_PATH = "src/test/resources/xmlsource/";
+  private static final String TARGET_FOLDER_PATH = "src/test/resources/xmltarget/";
+  private File sourceFolder = new File(SOURCE_FOLDER_PATH);
+  private File targetFolder = new File(TARGET_FOLDER_PATH);
+  private String sourceFolderUri = sourceFolder.toURI().toString();
+
+  @ClassRule
+  public static final TestConfiguration CONFIG = new TestConfiguration("explore.enabled", false);
+
+  @BeforeClass
+  public static void setupTest() throws Exception {
+    setupBatchArtifacts(BATCH_APP_ARTIFACT_ID, ETLBatchApplication.class);
+    addPluginArtifact(NamespaceId.DEFAULT.artifact("core-plugins", "1.0.1"), BATCH_APP_ARTIFACT_ID,
+                      XMLReaderBatchSource.class);
+  }
+
+  /**
+   * Method to copy test xml files into source folder path, from where XMLReader read the file.
+   */
+  private void copyFiles() {
+    try {
+      String xmlSourceFolder = "src/test/resources/";
+      String catalogLargeFile = "catalogLarge.xml";
+      String catalogSmallFile = "catalogSmall.xml";
+      FileUtils.copyFile(new File(xmlSourceFolder + catalogLargeFile), new File(SOURCE_FOLDER_PATH + catalogLargeFile));
+      FileUtils.copyFile(new File(xmlSourceFolder + catalogSmallFile), new File(SOURCE_FOLDER_PATH + catalogSmallFile));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Method to clear source and target folders. This ensures that source and target folder are ready to use for next
+   * JUnit Test case.
+   */
+  private void clearSourceAndTargetFolder() {
```

Comment:
Actually it is deleting, moving or archiving files after test case run and hence need to clear and restore files after every test case so that it will be available to next test case. Please suggest if it can be done in some better way.